doc/source/serve/tutorials/gradio-integration.md

```diff
@@ -0,0 +1,270 @@
+![Gradio Result](https://raw.githubusercontent.com/ray-project/ray/master/doc/source/serve/gradio_result.png)
+
+
+## Complex Example: Parallelism
+If you have multiple models you want to run in parallel, one way to do so is directly through Ray Serve. Suppose you want to run the following program: run two text generation models and print out both outputs.
+```python
+generator1 = pipeline("text-generation", model="gpt2")
+generator2 = pipeline("text-generation", model="EleutherAI/gpt-neo-125M")
+
+def model1(text):
+    generated_list = generator1(text, do_sample=True, min_length=20, max_length=100)
+    generated = generated_list[0]["generated_text"]
+    return generated
+
+def model2(text):
+    generated_list = generator2(text, do_sample=True, min_length=20, max_length=100)
+    generated = generated_list[0]["generated_text"]
+    return generated
+
+demo = gr.Interface(lambda text: f"{model1(text)}\n------------\n{model2(text)}", "textbox", "textbox")
+demo.launch()
+```
+
+We can parallelize the two text generation models by wrapping each model in a separate Ray Serve deployment.
+
+```python
+@serve.deployment
+class Text_Generation_Model:
+    def __init__(self, model_name):
+        self.generator = pipeline("text-generation", model=model_name)
+
+    def __call__(self, text):
+        generated_list = self.generator(text, do_sample=True, min_length=20, max_length=100)
+        generated = generated_list[0]["generated_text"]
+        return generated
+
+app1 = Text_Generation_Model.bind("gpt2")
+app2 = Text_Generation_Model.bind("EleutherAI/gpt-neo-125M")
+```
+
+Then, instead of simply wrapping our Gradio App in a Gradio Server deployment, we can reroute the Gradio App so that it runs the Text_Generation_Model deployments:
+
+```python
+@serve.deployment
+class GradioServer(GradioIngress):
+    def __init__(self, downstream_model_1, downstream_model_2):
+        self._d1 = downstream_model_1
+        self._d2 = downstream_model_2
+
+        io = gr.Interface(self.fanout, "textbox", "textbox")
+        super().__init__(io)
+
+    def fanout(self, text):
+        [result1, result2] = ray.get([self._d1.remote(text), self._d2.remote(text)])
+        return f"{result1}\n------------\n{result2}"
+```
+
+Lastly, we link everything together:
+```python
+app = GradioServer.bind(app1, app2)
+```
+
+This is the full script:
+```python
+from ray import serve
+import ray
+import gradio as gr
+from ray.serve.gradio import GradioIngress
+from transformers import pipeline
+
+
+@serve.deployment
+class GradioServer(GradioIngress):
+    def __init__(self, downstream_model_1, downstream_model_2):
+        self._d1 = downstream_model_1
+        self._d2 = downstream_model_2
+
+        io = gr.Interface(self.fanout, "textbox", "textbox")
+        super().__init__(io)
+
+    def fanout(self, text):
+        [result1, result2] = ray.get([self._d1.remote(text), self._d2.remote(text)])
+        return f"{result1}\n------------\n{result2}"
```

Comment:
nit: use `async def` as best practice :)

Reply:
Unfortunately Gradio doesn't support async functions in gr.Interface yet :(